test/web.js

```diff
@@ -313,4 +313,70 @@ lab.experiment("OAuth", function() {
       });
     });
   });
+
+  lab.test("GET access_token - invalid grant_type", function(done) {
+    ls.start(function(error) {
+      var accessTokenRequest = {
+        method: "GET",
+        url: "/login/oauth/access_token?client_id=test&client_secret=test&grant_type=client_credentials",
+        headers: {
+          "Content-Type": "application/x-www-form-urlencoded"
+        }
+      };
+
+      s.inject(authTokenRequest, function(authTokResponse) {
+        var redirectUri = url.parse(authTokResponse.headers.location, true);
+        accessTokenRequest.url += "&code=" + redirectUri.query.code;
+
+        s.inject(accessTokenRequest, function(response) {
+          Code.expect(response.statusCode).to.equal(400);
+          done();
+        });
+      });
+    });
+  });
+
+  lab.test("GET access_token - missing client_id", function(done) {
```

Comment:
I'm not sure of the value of testing two invalid error conditions here (missing client_id and wrong grant_type)